### ISSUE TITLE
How to train Deep Sort with Custom Dataset?

### ISSUE THREAD

**Issue by ImSuMyatNoe** [pmj110119/YOLOX_deepsort_tracker#8]
May I know about how to train Deep Sort with Custom Dataset?

**Comment by Sarouch** [comment]
@ImSuMyatNoe  There is a lot of things to change in the code to train on custom datasets, I'm lost...

**Comment by Peac12** [comment]
Sorry , DeepSORT not model for train , it just tracker , you should to train the model of yolo on your custom data set

**Comment by Sarouch** [comment]
> Sorry , DeepSORT not model for train , it just tracker , you should to train the model of yolo on your custom data set

@Peac12  Hello, Deepsort was trained on coco dataset, it is not just a tracker, it has a neural network inside, this is why it performs better than other trackers. (If I am wrong, please correct me)

**Comment by Peac12** [comment]
@Sarouch 
 I was searching about this methods That's used NN for Tracking , 
this methods is maded for MOT (Multi object tracking ) Not single object tracking  , 
and this algorithim like SORT & DeepSORT & ByteTrack  , it using Yolo for Detction object becuse it so fast and has high accuracy to  and Assign it to this method for tracking

**Comment by Peac12** [comment]
@Sarouch
you should  to seprate between Detection and Tracking 
Here , Detection using YOLO that was trained in coco or another dataset
you can train any model of yolo and take the weights after training to using in DeepSORT

**Comment by VCBE123** [comment]
Hello, could you please give me a tutorial on how to train the deep feature extractor? I can already run the tracker on my own dataset. But I need to finetune the Extractor. Any advice is a pleasure!

**Comment by Peac12** [comment]
Hello , I think you should to figure out how yolo works ? 
and try to train your custom data on any model of yolo ,then Notice his output , then you can to read the code of Deepsort and figure out , what doing yolo in deepsort

conclusion : I think your purpose respect with yolo , not Deepsort method , Deepsort just tracker not detection for your object ,
he send the coordinate of your object to your tracker(in our problem Deepsort) to track 

Sorry for my poor language because i am not a native speaker

**Comment by VCBE123** [comment]
@Peac12 
Hi, I already got my trained yolox. But in the Deepsort， there is a tiny network named Extractor trained on some datasets to get the embedding.  The extractor is used to compute the similarity between two boxes.

**Comment by Peac12** [comment]
@VCBE123  Hi, if you have a reference for this information ,plz tell me 
I aleardy read the paper of this algorithm https://arxiv.org/abs/2110.06864 and i don't read this the information that you metion

**Comment by Sarouch** [comment]
> @Sarouch you should to seprate between Detection and Tracking Here , Detection using YOLO that was trained in coco or another dataset you can train any model of yolo and take the weights after training to using in DeepSORT

Hello, thank you for your answers.
I used the weights obtained by yolo in which I have a very good accuracy !  but when I used deepSORT, the bbox were outside the object and the tracking was bad ! I had a lot of frames loss. Any way for this reason I wanted to retrain deepsort on my own datasets not on persons as it is the case now.

**Comment by umar-khayam** [comment]
Did you overcome with this problem?

**Comment by KyriakosChris** [comment]
Hello @VCBE123  I have the same problem as you did, I retrained yolov7 on Visdrone Dataset, since my problem is to detect vehicles from drones, but I want to retrain the deepsort tiny network, the Extractor. The best open source Dataset that I could get to retrain the deepsort is this [DroneCrowd](https://github.com/VisDrone/DroneCrowd) but I am not sure how to perform the training on the Extractor. Did you figure it out?

Thank you

**Comment by Hemilibeatriz** [comment]
Did you overcome with this problem?

**Comment by Ultios** [comment]
I have solved the problem of retraining deepsort with custom dataset.

As the deepsort implementation are from [ZQPei's Deepsort Pytorch implementation](https://github.com/ZQPei/deep_sort_pytorch), I used his retraining code to retrain my Deepsort checkpoint.
See his [tutorial file](https://github.com/ZQPei/deep_sort_pytorch/blob/master/deep_sort/deep/GETTING_STARTED.md) about the subject.

**Comment by Hemilibeatriz** [comment]
> Resolvi o problema de re-treinamento do Deep Sort com um conjunto de dados personalizado.
> 
> Como a implementação do Deepsort é baseada na [implementação do Deepsort em PyTorch do ZQPei](https://github.com/ZQPei/deep_sort_pytorch) , utilizei o código de retreinamento dele para retreinar meu checkpoint do Deepsort. Consulte [o arquivo tutorial](https://github.com/ZQPei/deep_sort_pytorch/blob/master/deep_sort/deep/GETTING_STARTED.md) dele para obter mais informações sobre o assunto.

are your results better?

**Comment by Ultios** [comment]
Yes, it is, by looking at the video.
I havent made evaluation metrics code yet for object tracking (hope someone gave me reference)

But according to my direct observation, it can track occluded object better than the deepsort checkpoint trained on pedestrian dataset. By a lot.

I can give you my jupyter notebook code for this particular problem of retraining.

**Comment by Hemilibeatriz** [comment]
> Yes, it is, by looking at the video. I havent made evaluation metrics code yet for object tracking (hope someone gave me reference)
> 
> But according to my direct observation, it can track occluded object better than the deepsort checkpoint trained on pedestrian dataset. By a lot.
> 
> I can give you my jupyter notebook code for this particular problem of retraining.

That's very kind and helpful of you. Could you share it with me at hemilitrindade@gmail.com:) I´d like to see the resulting videos
